Get list of all users

Noisyboy-9 opened the pull request:
One of the features was to get the list of all users, the route defined for this feature is GET /users.
I don't provide any unit testing at the moment cause I want to go through the code as fast as possible if there is time in the future I will add unit testing to make sure everything works fine!

Noisyboy-9 committed:
Remove unnecessary .gitkeep from folders

Noisyboy-9 committed:
Rename files to obate project naming conventions

Noisyboy-9 committed:
Add user.route.js

Noisyboy-9 committed:
Add users query for geting all users and getting user by id

Noisyboy-9 committed:
Finish users controller for getting list of all users

Noisyboy-9 committed:
Add getting list of all users to list of routes

Noisyboy-9 committed:
Change usersController.getAllUsers() res to have predefined structure

Noisyboy-9 committed:
Add auth check middleware

Noisyboy-9 committed:
Change get all users query to just get id,name,student_id,avatar of each user

arefehkompani commented on src/controllers/user.controller.js:
```diff
@@ -0,0 +1,15 @@
+    getAllUsers(req, res) {
+        const auth = new authMiddleware(req, res);
+
+        if (auth.checkToken()) {
```
`auth.checkToken()` doesn't show that our token is valid.
We need use `auth.getUser()` as a variable before functions to check the token.

arefehkompani commented on src/controllers/user.controller.js:
```diff
@@ -0,0 +1,15 @@
+        const auth = new authMiddleware(req, res);
+
+        if (auth.checkToken()) {
+            res.json({
```
As we talked about before, I think It's better to use services because from the query we receive "Promise" so we need `.then()` and `.catch()` here (you can see the examples in tarin or auth controller).
Also if `getAll()` has error we always see { }, but we need to response a related `message`.
![Screenshot (653)](https://user-images.githubusercontent.com/49534520/194931283-583d40f6-c4be-49bc-a625-084aeeaed7a1.png)

arefehkompani commented on src/controllers/user.controller.js:
```diff
@@ -0,0 +1,15 @@
+        if (auth.checkToken()) {
+            res.json({
+                'message': 'لیست تمام کاربران با موفقیت دریافت شد',
+                'users': getAll(),
```
I ran your code but I didn't get "users" in response.
![Screenshot (654)](https://user-images.githubusercontent.com/49534520/194932741-a59aaf48-5d9c-4e18-884a-770e14734adf.png)

I think it's because of the Promise but if you got response here please show me and lets talk about it.

Let's conceive we got all users information, for example if the user doesn't have the avatar we get _undefined_ which can cause the problem so we need to handle it and change it to _null_ value. So we need something like this in `res.json`:
`
avatar: user.avatar ? user.avatar : null
`

Noisyboy-9 committed:
Change checking for auth middleware usage

in the user.controller.js file there was the intention to check
for the user being authenticated and then doing the rest of
the logic which was done wrong, the getUser() function has to be called
whcih returns the auth user.

Noisyboy-9 committed:
Changed user.controller.js to use user.service.js

Noisyboy-9 committed:
Fix problems is user.service.js
